docs/transport.rst

```diff
@@ -219,3 +224,63 @@ In Serial Transport, the topic's packaging could be divided into two steps:
 The figure above shows the overhead added by Serial Transport.
 In the best case, it is **only 19 bytes**, but it should be noted that, in this example, the message stuffing has been neglected.
 
+Custom Transport API
+--------------------
+
+XRCE-DDS provides an user API that allows interfacing with the lowest level transport layer at runtime. This way an user is enabled to implement its own transports based on one of the two communication approaches: stream-oriented or packet-oriented.
+
+Client custom transport API
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By means of using the following functions an user can set four callbacks that will be in charge of opening and closing the transport, and writing and reading from it. This custom transport API is enabled by setting the CMake argument ``UCLIENT_PROFILE_CUSTOM_TRANSPORT=<bool>`` to true. In the case that stream-oriented transport is used ``UCLIENT_PROFILE_STREAM_FRAMING=<bool>`` should also be enabled.
+
+UXRDLLAPI void uxr_set_custom_transport_callbacks(uxrCustomTransport* transport,
+                                                  bool framing,
+                                                  open_custom_func open,
+                                                  close_custom_func close,
+                                                  write_custom_func write,
+                                                  read_custom_func read);
+------
+
+.. code-block:: c
+
+    void uxr_set_custom_transport_callbacks(uxrCustomTransport* transport, bool framing, open_custom_func open, close_custom_func close, write_custom_func write, read_custom_func read);
+
+Assigns the callback for custom transport.
+
```

Comment:
isn't the description of :transport: missing here?

Reply:
Solved